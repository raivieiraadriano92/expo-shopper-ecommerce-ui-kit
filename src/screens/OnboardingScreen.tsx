import { Box, Text } from 'native-base'
// import { WelcomeScreenTemplate } from '@shared'

import { RootStackScreenComponent } from '../navigation'

export const OnboardingScreen: RootStackScreenComponent<'Onboarding'> = ({ navigation }) => (
  <Box alignItems="center" bg="gray.300" flex={1} justifyContent="center" safeAreaTop>
    <Text>Cart</Text>
  </Box>
  // <WelcomeScreenTemplate
  //   goToRegisterScreen={() => navigation.navigate('Register')}
  //   goToSignInScreen={() => navigation.navigate('SignIn')}
  // />
)
