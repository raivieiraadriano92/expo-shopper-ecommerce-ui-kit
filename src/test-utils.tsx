import { FunctionComponent } from 'react'

import { render } from '@testing-library/react-native'

import { AppWrapper } from '../src/AppWrapper'

const Wrapper: FunctionComponent = ({ children }) => <AppWrapper>{children}</AppWrapper>

const customRender: typeof render = (ui, options) => render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
