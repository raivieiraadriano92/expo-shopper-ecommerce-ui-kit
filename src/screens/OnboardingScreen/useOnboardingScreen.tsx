import { useCallback, useMemo, useState } from 'react'

import { Illustrations } from 'src/components'
import { RootStackScreenComponentProps } from 'src/navigation'

const steps = [
  null,
  {
    heading: 'A wide range of goods',
    illustration: <Illustrations.Products size={327} />,
    text: 'Shopper allows customers to choose a product or service of their choice from any supplier anywhere in the world.'
  },
  {
    heading: 'Get anything pickup & delivered at your doorstep',
    illustration: <Illustrations.Delivery size={327} />,
    text: 'Order online to get groceries, medicines, laundry & may more delivered at doorstep.'
  },
  {
    heading: 'Create your own store',
    illustration: <Illustrations.OpenStore size={327} />,
    text: "Make it your own. As you're setting up your online store, you have the ability to customize your retail store online to start selling products."
  }
]

export const useOnboardingScreen = ({
  navigation
}: Pick<RootStackScreenComponentProps<'Onboarding'>, 'navigation'>) => {
  const [stepIndex, setStepIndex] = useState(0)

  const stepContent = useMemo(() => steps[stepIndex], [stepIndex])

  const goToWelcomeScreen = useCallback(() => navigation.replace('Welcome'), [navigation])

  const goToNextStep = useCallback(() => {
    if (stepIndex < 3) {
      setStepIndex((prev) => prev + 1)

      return
    }

    goToWelcomeScreen()
  }, [stepIndex, goToWelcomeScreen])

  return {
    goToNextStep,
    goToWelcomeScreen,
    stepContent,
    stepIndex
  }
}
