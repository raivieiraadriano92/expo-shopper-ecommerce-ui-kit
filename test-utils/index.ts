import { render } from '@testing-library/react-native'
import { AppWrapper } from 'src/AppWrapper'

const customRender: typeof render = (ui, options) => render(ui, { wrapper: AppWrapper, ...options })

// re-export everything
export * from '@testing-library/react-native'

// override render method
export { customRender as render }
