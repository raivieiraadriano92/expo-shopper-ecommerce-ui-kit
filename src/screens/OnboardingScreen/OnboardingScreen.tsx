import bgOnboarding from 'assets/bg-onboarding.png'
import { Button, Center, Container, Heading, HStack, Text, VStack } from 'native-base'
import { ImageBackground } from 'react-native'
import { Icons, Illustrations } from 'src/components'
import { RootStackScreenComponent } from 'src/navigation'

import { useOnboardingScreen } from './useOnboardingScreen'

export const OnboardingScreen: RootStackScreenComponent<'Onboarding'> = ({ navigation }) => {
  const { goToNextStep, goToWelcomeScreen, stepContent, stepIndex } = useOnboardingScreen({
    navigation
  })

  return (
    <ImageBackground resizeMode="contain" source={bgOnboarding} style={{ flex: 1 }}>
      <VStack {...(stepIndex ? { bg: 'white' } : {})} alignItems="center" flex={1} p={6} safeArea>
        <Container flex={1} maxW="md" w="full">
          {stepIndex ? (
            <>
              <VStack space={4} w="full">
                <Heading size="2xl">{stepContent?.heading}</Heading>
                <Text color="gray.500" variant="body2">
                  {stepContent?.text}
                </Text>
              </VStack>
              <Center flex={1} w="full">
                {stepContent?.illustration}
              </Center>
            </>
          ) : (
            <Center flex={1} w="full">
              <Illustrations.Logo size={238} />
            </Center>
          )}
          <HStack justifyContent="space-between" w="full">
            {!!stepIndex && (
              <Button colorScheme="secondary" onPress={goToWelcomeScreen} variant="link">
                Skip
              </Button>
            )}
            <Button
              {...(stepIndex ? { rightIcon: <Icons.ArrowRight size={5} /> } : { w: 'full' })}
              colorScheme="secondary"
              onPress={goToNextStep}>
              {stepIndex ? 'Next' : 'Explore Us'}
            </Button>
          </HStack>
        </Container>
      </VStack>
    </ImageBackground>
  )
}
