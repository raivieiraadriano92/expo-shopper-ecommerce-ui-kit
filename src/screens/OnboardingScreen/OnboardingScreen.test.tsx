import { render, waitFor } from 'test-utils'

import { OnboardingScreen } from './OnboardingScreen'

describe('<OnboardingScreen />', () => {
  it('renders correctly', async () => {
    //@ts-ignore
    const { findByText, toJSON } = render(<OnboardingScreen />)

    await waitFor(() => findByText('Explore Us'))

    expect(toJSON()).toMatchSnapshot()
  })
})
