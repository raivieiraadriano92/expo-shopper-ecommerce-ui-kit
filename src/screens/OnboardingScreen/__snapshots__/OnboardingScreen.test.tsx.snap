// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnboardingScreen /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "maxWidth": 448,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={52.721518987341774}
            bbWidth={238}
            fill="none"
            focusable={false}
            height={52.721518987341774}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 52,
                  "width": 238,
                },
              ]
            }
            vbHeight={35}
            vbWidth={158}
            width={238}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGRect
                fill={4278697893}
                height={32}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rx={6.526}
                width={32}
                x={0}
                y={1}
              />
              <RNSVGPath
                clipRule={0}
                d="M6.511 8.332a1.6 1.6 0 10-1.067.078c.748 5.16 5.19 9.123 10.556 9.123 5.367 0 9.808-3.963 10.556-9.123a1.6 1.6 0 10-1.067-.078c-.706 4.606-4.685 8.135-9.49 8.135-4.803 0-8.783-3.529-9.488-8.135z"
                fill={4294967295}
                fillRule={0}
                propList={
                  Array [
                    "fill",
                    "fillRule",
                  ]
                }
              />
              <RNSVGPath
                d="M52.216 27.182c-1.335 0-2.53-.217-3.588-.65-1.057-.433-1.907-1.075-2.548-1.924-.624-.85-.953-1.872-.988-3.068h4.732c.07.676.303 1.196.702 1.56.399.347.919.52 1.56.52.659 0 1.179-.147 1.56-.442.381-.312.572-.737.572-1.274 0-.45-.156-.823-.468-1.118a3.47 3.47 0 00-1.118-.728c-.433-.19-1.057-.407-1.872-.65-1.179-.364-2.14-.728-2.886-1.092a5.332 5.332 0 01-1.924-1.612c-.537-.71-.806-1.638-.806-2.782 0-1.699.615-3.025 1.846-3.978 1.23-.97 2.834-1.456 4.81-1.456 2.01 0 3.631.485 4.862 1.456 1.23.953 1.89 2.288 1.976 4.004h-4.81c-.035-.59-.251-1.049-.65-1.378-.399-.347-.91-.52-1.534-.52-.537 0-.97.147-1.3.442-.33.277-.494.685-.494 1.222 0 .59.277 1.049.832 1.378.555.33 1.421.685 2.6 1.066 1.179.399 2.132.78 2.86 1.144a5.435 5.435 0 011.924 1.586c.537.693.806 1.586.806 2.678 0 1.04-.269 1.985-.806 2.834-.52.85-1.283 1.525-2.288 2.028-1.005.503-2.193.754-3.562.754zm18.31-14.846c1.664 0 2.999.555 4.004 1.664 1.005 1.092 1.508 2.6 1.508 4.524V27h-4.42v-7.878c0-.97-.251-1.725-.754-2.262-.503-.537-1.179-.806-2.028-.806-.85 0-1.525.269-2.028.806-.503.537-.754 1.291-.754 2.262V27h-4.446V7.76h4.446v6.682c.45-.641 1.066-1.153 1.846-1.534s1.655-.572 2.626-.572zm15.206 14.872c-1.422 0-2.704-.303-3.848-.91a6.774 6.774 0 01-2.678-2.6c-.642-1.127-.962-2.444-.962-3.952 0-1.49.329-2.8.988-3.926a6.68 6.68 0 012.704-2.626c1.144-.607 2.426-.91 3.848-.91 1.421 0 2.704.303 3.848.91a6.68 6.68 0 012.704 2.626c.658 1.127.988 2.435.988 3.926 0 1.49-.338 2.808-1.014 3.952a6.686 6.686 0 01-2.73 2.6c-1.144.607-2.427.91-3.848.91zm0-3.848c.849 0 1.568-.312 2.158-.936.606-.624.91-1.517.91-2.678s-.295-2.054-.884-2.678a2.774 2.774 0 00-2.132-.936c-.867 0-1.586.312-2.158.936-.572.607-.858 1.5-.858 2.678 0 1.161.277 2.054.832 2.678a2.775 2.775 0 002.132.936zm14.396-8.814c.434-.676 1.032-1.222 1.794-1.638.763-.416 1.656-.624 2.678-.624 1.196 0 2.28.303 3.25.91.971.607 1.734 1.473 2.288 2.6.572 1.127.858 2.435.858 3.926 0 1.49-.286 2.808-.858 3.952-.554 1.127-1.317 2.002-2.288 2.626-.97.607-2.054.91-3.25.91-1.005 0-1.898-.208-2.678-.624-.762-.416-1.36-.953-1.794-1.612v8.944h-4.446V12.492h4.446v2.054zm6.344 5.174c0-1.11-.312-1.976-.936-2.6-.606-.641-1.36-.962-2.262-.962-.884 0-1.638.32-2.262.962-.606.641-.91 1.517-.91 2.626 0 1.11.304 1.985.91 2.626.624.641 1.378.962 2.262.962.884 0 1.638-.32 2.262-.962.624-.659.936-1.543.936-2.652z"
                fill={4278256653}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
              <RNSVGPath
                d="M117.775 14.546c.433-.676 1.031-1.222 1.794-1.638.762-.416 1.655-.624 2.678-.624 1.196 0 2.279.303 3.25.91.97.607 1.733 1.473 2.288 2.6.572 1.127.858 2.435.858 3.926 0 1.49-.286 2.808-.858 3.952-.555 1.127-1.318 2.002-2.288 2.626-.971.607-2.054.91-3.25.91-1.006 0-1.898-.208-2.678-.624-.763-.416-1.361-.953-1.794-1.612v8.944h-4.446V12.492h4.446v2.054zm6.344 5.174c0-1.11-.312-1.976-.936-2.6-.607-.641-1.361-.962-2.262-.962-.884 0-1.638.32-2.262.962-.607.641-.91 1.517-.91 2.626 0 1.11.303 1.985.91 2.626.624.641 1.378.962 2.262.962.884 0 1.638-.32 2.262-.962.624-.659.936-1.543.936-2.652zm20.532-.208c0 .416-.026.85-.078 1.3h-10.062c.07.901.356 1.595.858 2.08.52.468 1.153.702 1.898.702 1.11 0 1.881-.468 2.314-1.404h4.732a6.393 6.393 0 01-1.326 2.574 6.448 6.448 0 01-2.366 1.794c-.953.433-2.019.65-3.198.65-1.421 0-2.686-.303-3.796-.91a6.475 6.475 0 01-2.6-2.6c-.624-1.127-.936-2.444-.936-3.952s.304-2.825.91-3.952a6.475 6.475 0 012.6-2.6c1.11-.607 2.384-.91 3.822-.91 1.404 0 2.652.295 3.744.884a6.259 6.259 0 012.548 2.522c.624 1.092.936 2.366.936 3.822zm-4.55-1.17c0-.763-.26-1.37-.78-1.82-.52-.45-1.17-.676-1.95-.676-.745 0-1.378.217-1.898.65-.502.433-.814 1.049-.936 1.846h5.564zm11.342-3.432a5.743 5.743 0 011.95-1.872 4.955 4.955 0 012.6-.702v4.706h-1.222c-1.11 0-1.942.243-2.496.728-.555.468-.832 1.3-.832 2.496V27h-4.446V12.492h4.446v2.418z"
                fill={4278697893}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            _disabled={
              Object {
                "bg": "trueGray.300",
                "opacity": "50",
              }
            }
            _loading={
              Object {
                "bg": "warmGray.50",
                "opacity": "50",
              }
            }
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="secondary"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#01040D",
                "borderRadius": 16,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 24,
                "paddingRight": 24,
                "paddingTop": 8,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={Object {}}
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#FFFFFF",
                      "fontFamily": "Poppins_700Bold",
                      "fontSize": 16,
                      "fontStyle": undefined,
                      "fontWeight": undefined,
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Explore Us
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
